qcodes/instrument_drivers/QDev/QDac_channels.py

```diff
@@ -0,0 +1,684 @@
+                        '{} s'.format(ramptime) + '. Voltage not changed.')
+
+        offset = v_start
+        amplitude = setvoltage-v_start
+        if self.channels[chan-1].vrange.get_latest() == 1:
+            offset *= 10
+            amplitude *= 10
+
+        chanmssg = 'wav {} {} {} {}'.format(chan, fg,
+                                            amplitude,
+                                            offset)
+
+        if chan in [syn[0] for syn in self._syncoutputs]:
+            #syncing = True
+            sync = [syn[1] for syn in self._syncoutputs if syn[0] == chan][0]
+            sync_duration = 1000*self.channels[chan-1].sync_duration.get()
+            sync_delay = 1000*self.channels[chan-1].sync_delay.get()
+            self.write('syn {} {} {} {}'.format(sync, fg,
+                                                sync_delay,
+                                                sync_duration))
+        else:
+            syncing = False
+
+        typedict = {'SINE': 1, 'SQUARE': 2, 'RAMP': 3}
+
+        typeval = typedict['RAMP']
+        dutyval = 100
+        # s -> ms
+        periodval = ramptime*1e3
+        repval = 1
+        funmssg = 'fun {} {} {} {} {}'.format(fg,
+                                              typeval, periodval,
+                                              dutyval, repval)
+        self.write(chanmssg)
+        self.write(funmssg)
+
+    def write(self, cmd):
+        """
+        QDac always returns something even from set commands, even when
+        verbose mode is off, so we'll override write to take this out
+        if you want to use this response, we put it in self._write_response
+        (but only for the very last write call)
+
+        Note that this procedure makes it cumbersome to handle the returned
+        messages from concatenated commands, e.g. 'wav 1 1 1 0;fun 2 1 100 1 1'
+        Please don't use concatenated commands
+
+        TODO (WilliamHPNielsen): add automatic de-concatenation of commands.
+        """
+        if self.debugmode:
+            log.info('Sending command string: {}'.format(cmd))
+
+        nr_bytes_written, ret_code = self.visa_handle.write(cmd)
+        self.check_error(ret_code)
+        self._write_response = self.visa_handle.read()
+
+    def read(self):
+        return self.visa_handle.read()
+
+    def _wait_and_clear(self, delay=0.5):
+        time.sleep(delay)
+        self.visa_handle.clear()
+
+    def connect_message(self):
+        """
+        Override of the standard Instrument class connect_message.
+        Usually, the response to `*IDN?` is printed. Here, the
+        software version is printed.
+        """
+        self.visa_handle.write('status')
+
+        log.info('Connected to QDac on {}, {}'.format(self._address,
+                                                      self.visa_handle.read()))
+
+        # take care of the rest of the output
+        for ii in range(50):
```

Comment:
Or maybe not... It's not so clear *why* this output is exactly 50 lines long (firmware design choice), but it will not change with `self.num_chans`. But I agree that magic numbers should be avoided.